### ISSUE TITLE
Linebreak issue with tooltip display of function comment

### ISSUE THREAD

**Issue by Granine** [microsoft/pylance-release#4074]
Type: <b>Bug</b>

# Behaviour
![image](https://user-images.githubusercontent.com/89324736/225165278-9a079c94-6b93-491d-9273-e0452b7935ca.png)

## Expected vs. Actual
Expected: linebreak displayed for every line containing a colon ":"
Actual: linebreak only displayed for lines with an odd number of  colon ":"

## Steps to reproduce:
copy and paste the following demo function in a .py file and hover over call to demo()
```
def demo():
    '''
    c I will not line break,
    c because i have no colon.

    c I will line break,
    c: because i have colon,
    c::: odd number of colon.

    c I should line break,
    c:: because i have colon,
    c:::: but I am not, because I have even number of colon.
    '''
    return 0
demo()
```

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.7.8
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Global
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

experiments
• enabled: false

```

</p>
</details>

Extension version: 2023.3.10
VS Code version: Code 1.75.1 (441438abd1ac652551dbe4d408dfcec8a499b8bf, 2023-02-08T21:32:34.589Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: No

**Comment by rchiodo** [comment]
Thanks for the issue but I believe this is on purpose:

There's no standard for how doc strings should be split and we're just making it behave this way with an odd number of colons.
https://github.com/microsoft/pyright/blob/c272a608f3d6c25ea35ca8c67c22655b988f70a9/packages/pyright-internal/src/analyzer/docStringConversion.ts#L587

Do you have a use case where an even number of colons should provide a line break or did you just notice it behaves this way?

**Comment by Granine** [comment]
An example of comment
```
        @param `original:str or original:list`: original string to compare
        @param `new:str`: the new string to compare
        @return `int`: levenshtein difference
        @return `list`: levenshtein difference if list
```
in this case, we are attaching types to variables,causing some lines to have duplicates
(normally hinted in the function declaration, but here the function takes multiple types) and so we need to place it in the comment. 
Clearly, there are workarounds like `var(type)`: or `var`: *then explain in natural language the type intake*, but it is weird that only an even number of colons is accepted.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.4.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023431-19-April-2023-PreRelease)
